### ISSUE TITLE
istioctl analyze cannot handle a List

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#19880]
What I want to do: `kubectl get -n istio-system envoyfilter  -oyaml | ik analyze -`

I am met with an obscure error:
```
Error(s) adding files: 1 error occurred:
        * errors parsing content "\x00": 1 error occurred:
        * error processing [0]: failed finding spec for group/kind: /List



No validation issues found (but 1 file could not be parsed)
Error: Some files couldn't be parsed.
```

because we can't process a List.

Full config if needed for reproduction:

<details>

```yaml
apiVersion: v1
items:
- apiVersion: networking.istio.io/v1alpha3
  kind: EnvoyFilter
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"networking.istio.io/v1alpha3","kind":"EnvoyFilter","metadata":{"annotations":{},"labels":{"operator.istio.io/component":"Pilot","operator.istio.io/managed":"Reconcile","operator.istio.io/version":"1.4.3"},"name":"metadata-exchange-1.4","namespace":"istio-system"},"spec":{"configPatches":[{"applyTo":"HTTP_FILTER","match":{"context":"ANY","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager"}}},"proxy":{"proxyVersion":"1\\.4.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"envoy.wasm.metadata_exchange","vm_config":{"code":{"inline_string":"envoy.wasm.metadata_exchange"},"runtime":"envoy.wasm.runtime.null"}}},"name":"envoy.filters.http.wasm"}}}]}}
    creationTimestamp: "2020-01-01T17:53:14Z"
    generation: 1
    labels:
      operator.istio.io/component: Pilot
      operator.istio.io/managed: Reconcile
      operator.istio.io/version: 1.4.3
    name: metadata-exchange-1.4
    namespace: istio-system
    resourceVersion: "598"
    selfLink: /apis/networking.istio.io/v1alpha3/namespaces/istio-system/envoyfilters/metadata-exchange-1.4
    uid: f9688e6e-e751-4095-8496-a3ce65dbb5f0
  spec:
    configPatches:
    - applyTo: HTTP_FILTER
      match:
        context: ANY
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
        proxy:
          proxyVersion: 1\.4.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: envoy.wasm.metadata_exchange
              vm_config:
                code:
                  inline_string: envoy.wasm.metadata_exchange
                runtime: envoy.wasm.runtime.null
          name: envoy.filters.http.wasm
- apiVersion: networking.istio.io/v1alpha3
  kind: EnvoyFilter
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"networking.istio.io/v1alpha3","kind":"EnvoyFilter","metadata":{"annotations":{},"labels":{"operator.istio.io/component":"Pilot","operator.istio.io/managed":"Reconcile","operator.istio.io/version":"1.4.3"},"name":"metadata-exchange-1.5","namespace":"istio-system"},"spec":{"configPatches":[{"applyTo":"HTTP_FILTER","match":{"context":"ANY","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager"}}},"proxy":{"proxyVersion":"1\\.5.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"envoy.wasm.metadata_exchange","vm_config":{"code":{"local":{"inline_string":"envoy.wasm.metadata_exchange"}},"runtime":"envoy.wasm.runtime.null"}}},"name":"envoy.filters.http.wasm"}}}]}}
    creationTimestamp: "2020-01-01T17:53:14Z"
    generation: 1
    labels:
      operator.istio.io/component: Pilot
      operator.istio.io/managed: Reconcile
      operator.istio.io/version: 1.4.3
    name: metadata-exchange-1.5
    namespace: istio-system
    resourceVersion: "599"
    selfLink: /apis/networking.istio.io/v1alpha3/namespaces/istio-system/envoyfilters/metadata-exchange-1.5
    uid: c3f8911e-f78d-4ff7-8f16-36b88c884697
  spec:
    configPatches:
    - applyTo: HTTP_FILTER
      match:
        context: ANY
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
        proxy:
          proxyVersion: 1\.5.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: envoy.wasm.metadata_exchange
              vm_config:
                code:
                  local:
                    inline_string: envoy.wasm.metadata_exchange
                runtime: envoy.wasm.runtime.null
          name: envoy.filters.http.wasm
- apiVersion: networking.istio.io/v1alpha3
  kind: EnvoyFilter
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"networking.istio.io/v1alpha3","kind":"EnvoyFilter","metadata":{"annotations":{},"labels":{"operator.istio.io/component":"Pilot","operator.istio.io/managed":"Reconcile","operator.istio.io/version":"1.4.3"},"name":"stats-filter-1.4","namespace":"istio-system"},"spec":{"configPatches":[{"applyTo":"HTTP_FILTER","match":{"context":"SIDECAR_OUTBOUND","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.4.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_outbound","vm_config":{"code":{"inline_string":"envoy.wasm.stats"},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_outbound"}}},"name":"envoy.filters.http.wasm"}}},{"applyTo":"HTTP_FILTER","match":{"context":"SIDECAR_INBOUND","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.4.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_inbound","vm_config":{"code":{"inline_string":"envoy.wasm.stats"},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_inbound"}}},"name":"envoy.filters.http.wasm"}}},{"applyTo":"HTTP_FILTER","match":{"context":"GATEWAY","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.4.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_outbound","vm_config":{"code":{"inline_string":"envoy.wasm.stats"},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_outbound"}}},"name":"envoy.filters.http.wasm"}}}]}}
    creationTimestamp: "2020-01-01T17:53:14Z"
    generation: 1
    labels:
      operator.istio.io/component: Pilot
      operator.istio.io/managed: Reconcile
      operator.istio.io/version: 1.4.3
    name: stats-filter-1.4
    namespace: istio-system
    resourceVersion: "600"
    selfLink: /apis/networking.istio.io/v1alpha3/namespaces/istio-system/envoyfilters/stats-filter-1.4
    uid: 551da053-2207-489b-8924-ca072c200744
  spec:
    configPatches:
    - applyTo: HTTP_FILTER
      match:
        context: SIDECAR_OUTBOUND
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.4.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_outbound
              vm_config:
                code:
                  inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_outbound
          name: envoy.filters.http.wasm
    - applyTo: HTTP_FILTER
      match:
        context: SIDECAR_INBOUND
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.4.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_inbound
              vm_config:
                code:
                  inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_inbound
          name: envoy.filters.http.wasm
    - applyTo: HTTP_FILTER
      match:
        context: GATEWAY
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.4.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_outbound
              vm_config:
                code:
                  inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_outbound
          name: envoy.filters.http.wasm
- apiVersion: networking.istio.io/v1alpha3
  kind: EnvoyFilter
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"networking.istio.io/v1alpha3","kind":"EnvoyFilter","metadata":{"annotations":{},"labels":{"operator.istio.io/component":"Pilot","operator.istio.io/managed":"Reconcile","operator.istio.io/version":"1.4.3"},"name":"stats-filter-1.5","namespace":"istio-system"},"spec":{"configPatches":[{"applyTo":"HTTP_FILTER","match":{"context":"SIDECAR_OUTBOUND","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.5.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_outbound","vm_config":{"code":{"local":{"inline_string":"envoy.wasm.stats"}},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_outbound"}}},"name":"envoy.filters.http.wasm"}}},{"applyTo":"HTTP_FILTER","match":{"context":"SIDECAR_INBOUND","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.5.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_inbound","vm_config":{"code":{"local":{"inline_string":"envoy.wasm.stats"}},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_inbound"}}},"name":"envoy.filters.http.wasm"}}},{"applyTo":"HTTP_FILTER","match":{"context":"GATEWAY","listener":{"filterChain":{"filter":{"name":"envoy.http_connection_manager","subFilter":{"name":"envoy.router"}}}},"proxy":{"proxyVersion":"1\\.5.*"}},"patch":{"operation":"INSERT_BEFORE","value":{"config":{"config":{"configuration":"{\n  \"debug\": \"false\",\n  \"stat_prefix\": \"istio\",\n}\n","root_id":"stats_outbound","vm_config":{"code":{"local":{"inline_string":"envoy.wasm.stats"}},"runtime":"envoy.wasm.runtime.null","vm_id":"stats_outbound"}}},"name":"envoy.filters.http.wasm"}}}]}}
    creationTimestamp: "2020-01-01T17:53:14Z"
    generation: 1
    labels:
      operator.istio.io/component: Pilot
      operator.istio.io/managed: Reconcile
      operator.istio.io/version: 1.4.3
    name: stats-filter-1.5
    namespace: istio-system
    resourceVersion: "601"
    selfLink: /apis/networking.istio.io/v1alpha3/namespaces/istio-system/envoyfilters/stats-filter-1.5
    uid: 03e40c66-8e3a-4844-beb9-4cf86b7e3c4d
  spec:
    configPatches:
    - applyTo: HTTP_FILTER
      match:
        context: SIDECAR_OUTBOUND
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.5.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_outbound
              vm_config:
                code:
                  local:
                    inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_outbound
          name: envoy.filters.http.wasm
    - applyTo: HTTP_FILTER
      match:
        context: SIDECAR_INBOUND
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.5.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_inbound
              vm_config:
                code:
                  local:
                    inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_inbound
          name: envoy.filters.http.wasm
    - applyTo: HTTP_FILTER
      match:
        context: GATEWAY
        listener:
          filterChain:
            filter:
              name: envoy.http_connection_manager
              subFilter:
                name: envoy.router
        proxy:
          proxyVersion: 1\.5.*
      patch:
        operation: INSERT_BEFORE
        value:
          config:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                }
              root_id: stats_outbound
              vm_config:
                code:
                  local:
                    inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_outbound
          name: envoy.filters.http.wasm
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

</details>

**Comment by sushicw** [comment]
More specifically, our [in-memory kube source](https://github.com/istio/istio/blob/5f6e69b9338697f902508cc44246483bfabb4ad3/galley/pkg/config/source/kube/inmemory/kubesource.go) can't handle a list. 

This is theoretically broader than analyze, although I don't think any other uses are meaningfully affected. It's also used for various test paths, and for reading the mesh config as a file.

**Comment by esnible** [comment]
@selmanj This seems to be fixed in Master, but not in 1.5.  Shall we close this issue?

**Comment by selmanj** [comment]
This should be fixed by #21087, which was just merged into the 1.5 branch.

Note that it was fixed merely by ignoring types we don't support rather than complaining loudly.
